packages/@ember/-internals/routing/lib/system/router.ts

```diff
@@ -380,50 +371,22 @@ class EmberRouter extends EmberObject.extend(Evented) implements Evented {
         });
       }
 
+      // TODO: merge into routeDidChange
       didTransition(infos: PrivateRouteInfo[]) {
-        if (ROUTER_EVENTS) {
-          if (router.didTransition !== defaultDidTransition) {
-            deprecate(
-              'You attempted to override the "didTransition" method which is deprecated. Please inject the router service and listen to the "routeDidChange" event.',
-              false,
-              {
-                id: 'deprecate-router-events',
-                until: '4.0.0',
-                url: 'https://deprecations.emberjs.com/v3.x#toc_deprecate-router-events',
-                for: 'ember-source',
-                since: {
-                  enabled: '3.11.0',
-                },
-              }
-            );
-          }
-        }
+        assert(
+          'You attempted to override the "didTransition" method which has been deprecated. Please inject the router service and listen to the "routeDidChange" event.',
+          router.didTransition === defaultDidTransition
+        );
         router.didTransition(infos);
```

Comment:
I'm wondering if we can take this a step further and make defaultDidTransition && defaultWillTransition the defacto method that is called when didTransition/willTransition is called on the PrivateRouter. I'm wondering if we can take everything in, say defaultDidTransition and just throw it in here and then get rid of the `assert`.